src/Templates/BlockTemplatesCompatibility.php

```diff
@@ -0,0 +1,277 @@
+ *
+ * @internal
+ */
+class BlockTemplatesCompatibility {
+
+	const LOOP_ITEM_ID = 'product-loop-item';
+
+	/**
+	 * Hoding the supported hooks, their positions and the hooked functions.
+	 *
+	 * @var array $hook_data
+	 */
+	protected $hook_data = array(
+		'core/query'            => array(
+			'woocommerce_before_main_content' => array(
+				'position' => 'before',
+				'hooked'   => array(
+					'woocommerce_output_content_wrapper' => 10,
+					'woocommerce_breadcrumb'             => 20,
+				),
+			),
+			'woocommerce_after_main_content'  => array(
+				'position' => 'after',
+				'hooked'   => array(
+					'woocommerce_output_content_wrapper_end' => 10,
+				),
+			),
+		),
+		'core/post-title'       => array(
+			'woocommerce_before_shop_loop_item_title' => array(
+				'position' => 'before',
+				'hooked'   => array(
+					'woocommerce_show_product_loop_sale_flash' => 10,
+					'woocommerce_template_loop_product_thumbnail' => 10,
+				),
+			),
+			'woocommerce_shop_loop_item_title'        => array(
+				'position' => 'after',
+				'hooked'   => array(
+					'woocommerce_template_loop_product_title' => 10,
+				),
+			),
+			'woocommerce_after_shop_loop_item_title'  => array(
+				'position' => 'after',
+				'hooked'   => array(
+					'woocommerce_template_loop_rating' => 5,
+					'woocommerce_template_loop_price'  => 10,
+				),
+			),
+		),
+		self::LOOP_ITEM_ID      => array(
+			'woocommerce_before_shop_loop_item' => array(
+				'position' => 'before',
+				'hooked'   => array(
+					'woocommerce_template_loop_product_link_open' => 10,
+				),
+			),
+			'woocommerce_after_shop_loop_item'  => array(
+				'position' => 'after',
+				'hooked'   => array(
+					'woocommerce_template_loop_product_link_close' => 5,
+					'woocommerce_template_loop_add_to_cart' => 10,
+				),
+			),
+		),
+		'core/post-template'    => array(
+			'woocommerce_before_shop_loop' => array(
+				'position' => 'before',
+				'hooked'   => array(
+					'woocommerce_output_all_notices' => 10,
+					'woocommerce_result_count'       => 20,
+					'woocommerce_catalog_ordering'   => 30,
+				),
+			),
+			'woocommerce_after_shop_loop'  => array(
+				'position' => 'after',
+				'hooked'   => array(
+					'woocommerce_pagination' => 10,
+				),
+			),
+		),
+		'core/query-no-results' => array(
+			'woocommerce_no_products_found' => array(
+				'position' => 'before',
+				'hooked'   => array(
+					'wc_no_products_found' => 10,
+				),
+			),
+		),
+	);
+
+	/**
+	 * Constructor.
+	 */
+	public function __construct() {
+		$this->init();
+	}
+
+	/**
+	 * Initialization method.
+	 */
+	protected function init() {
+		if ( ! wc_current_theme_is_fse_theme() ) {
+			return;
+		}
+
+		add_filter( 'render_block_data', array( $this, 'update_render_block_data' ), 10, 3 );
+		add_filter( 'render_block', array( $this, 'inject_hooks' ), 10, 2 );
+		add_action( 'template_redirect', array( $this, 'remove_default_hooks' ) );
```

Comment:
This assumes that the _products block_ is powering the archive template right? You'll probably need to add some detection of the classic template block to cover:

- being able to merge this without having converted the default templates to be powered by the products block.
- themes that may have overridden the Woo block templates (which would still be using the classic block)
- custom user saved templates that are using the classic block.

Reply:
I completely forgot the Classic Template Block. Thanks!